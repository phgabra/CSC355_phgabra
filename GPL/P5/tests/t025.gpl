
int b = 1;

int c = !b;
int d = -c;

int e = -b;
int f = !b;

int g = !-b;
int h = -!b;

int i = !(-b);
int j = -(!b);

// ! should have a higher precedence than * resulting in 0
// If the precedences are reverse the result would be 1
int k = !b * 0;

