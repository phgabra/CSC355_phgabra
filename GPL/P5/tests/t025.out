gpl.cpp::main()
  input file(tests/t025.gpl)
  random seed(42)
  read_keypresses_from_standard_input(false)
  dump_pixels(false)
  symbol_table(true)
  graphics(false)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().

No errors found (parser probably worked correctly).


Printing the symbol table from main()
int b = 1
int c = 0
int d = 0
int e = -1
int f = 0
int g = 0
int h = 0
int i = 0
int j = 0
int k = 0
Graphics is turned off by the Makefile.  Program exiting.

gpl.cpp::main() done.
